nautobot/extras/tests/test_models.py

```diff
@@ -1319,6 +1322,55 @@ def get_absolute_url(self):
         self.assertEqual("", log.absolute_url)
 
 
+class ObjectMetadataTest(ModelTestCases.BaseModelTestCase):
+    model = ObjectMetadata
+
+    def test_immutable_metadata_type(self):
+        instance1 = ObjectMetadata.objects.first()
+        instance2 = ObjectMetadata.objects.exclude(metadata_type=instance1.metadata_type).first()
+        self.assertIsNotNone(instance2)
+        with self.assertRaises(ValidationError):
+            instance1.metadata_type = instance2.metadata_type
+            instance1.validated_save()
+
```

Comment:
Would want to see `value` validation tests here, similar to `test_customfields.py`.